base/socket.jl

```diff
@@ -245,6 +245,7 @@ type InetAddr
     end
 end
 
+Base.call{T<:IPAddr}(::Type{InetAddr}, ip::T, port) = InetAddr{T}(ip, port)
```

Comment:
style point: this constructor would be clearer written without the explicit call:
```julia
InetAddr{T<:IPAddr}(ip::T, port) = InetAddr{T}(ip, port)
or equivalently
InetAddr(ip::IPAddr, port) = InetAddr{typeof(ip)}(ip, port)
```